dotcom-rendering/src/web/components/CommercialMetrics.tsx

```diff
@@ -31,16 +37,21 @@ export const CommercialMetrics: React.FC<{
 			window.guardian.config.page.isDev ||
 			window.location.hostname.includes('localhost');
 
-		if (isDev || shouldForceMetrics || userIsInSamplingGroup)
+		if (isDev || shouldForceMetrics || userIsInSamplingGroup) {
+			const args: [string, string | undefined, boolean, boolean?] = adBlockerInUse === undefined
+				? [pageViewId, browserId, Boolean(isDev)]
+				: [pageViewId, browserId, Boolean(isDev), adBlockerInUse];
 			setSentCommercialMetrics(
-				sendCommercialMetrics(pageViewId, browserId, Boolean(isDev)),
+				sendCommercialMetrics(...args),
```

Comment:
I think that, depending on the order of things, this code might end up being run more than once. The first time this code is run `adBlockerInUse` will be undefined and then - if an ad blocker is found - it will get set to `true`, triggering this code to rerun.

I wonder if moving over to `useOnce` now might be easiest here? It's actually pretty straight forward refactor and that way it guarantees we only send metrics once.  Happy to pair on this if it helps?